sirepo/pkcli/admin.py

```diff
@@ -77,44 +77,60 @@ def move_user_sims(target_uid=''):
         shutil.move(lib_file, target)
 
 
-def purge_guest_users(days=180, confirm=False):
-    """Remove old users from db which have not registered.
+def purge_free_users(days=180, confirm=False):
+    """Remove basic users simulations
 
     Args:
         days (int): maximum days of untouched files (old is mtime > days)
         confirm (bool): delete the directories if True (else don't delete) [False]
 
     Returns:
-        (list, list): dirs and uids of removed guest users (or to remove if confirm)
+        (list, list): dirs and uids of removed users (or to remove if confirm)
     """
 
+    def _check_expired_run_dir(path):
+        d = str(path).lower()
+        if 'report' in d or 'animation' in d:
+            for f in pkio.walk_tree(path):
+                # https://github.com/radiasoft/sirepo/issues/1888
+                if not f.exists():
+                    continue
+                if (n - n.fromtimestamp(f.mtime())).days <= days:
+                    return
+            else:
+                res.append(str(path))
+
+    def _get_premium_uids():
+        r = auth_db.UserRole
+        return [
+            x[0] for x in r.query.with_entities(r.uid).filter(
+                r.role == auth.ROLE_PREMIUM,
+            ).distinct().all()
+        ]
+
     days = int(days)
     assert days >= 1, \
         '{}: days must be a positive integer'
     server.init()
     from sirepo import srtime
 
-    guest_uids = auth.guest_uids()
-    now = srtime.utc_now()
-    dirs_and_uids = {}
-
-    for d in pkio.sorted_glob(simulation_db.user_dir_name().join('*')):
-        uid = simulation_db.uid_from_dir_name(d)
-        if _is_src_dir(d):
-            continue
-        if uid not in guest_uids:
-            continue
-        for f in pkio.walk_tree(d):
-            if (now - now.fromtimestamp(f.mtime())).days <= days:
-                break
-        else:
-
-            dirs_and_uids[d] = uid
-    if confirm:
-        pkio.unchecked_remove(*dirs_and_uids.keys())
-        auth_db.UserRegistration.delete_all_for_column_by_values('uid', dirs_and_uids.values())
-
-    return dirs_and_uids
+    with auth_db.thread_lock:
+        u = _get_premium_uids()
+        n = srtime.utc_now()
+        res = []
+        for p in pkio.sorted_glob(simulation_db.user_dir_name().join(
```

Comment:
Let's add this abstraction to simulation_db, which controls the structure of simulation run dirs.